### DISCUSSION TITLE
使用Message和MessageSegment时出现Can't instantiate abstract class Message/MessageSegment with abstract methods问题

### DISCUSSION THREAD

**Discussion by TDK1969** [nonebot/discussions#108] [问答] [answered]
环境：Ubuntu
版本：Nonebot2
问题描述：想使用通过bot.send_msg发送图片，通过查询指南网站得知可以通过Message和MessageSegment类发送图片，如下：
![image](https://user-images.githubusercontent.com/54183084/155311902-72af0f7d-4f19-433f-8baf-52dcf0ff0681.png)
模仿上述代码，以下是我的代码
```
schedBot = nonebot.get_bot()
coverMsg = MessageSegment(type='image', data={'url':coverURL})
reportedMsg = Message([
       MessageSegment(type='text', data={'text':textMsg}),
       MessageSegment(type='image', data={'url':coverURL})
])
await schedBot.send_msg(message=reportedMsg, user_id=myqq)
```

运行时，发生如下错误：
Can't instantiate abstract class MessageSegment with abstract methods __str__, get_message_class, is_textTraceback (most recent call last):
  File "/root/project/NoneBot/TDK_Bot/src/plugins/bilibili.py", line 109, in CheckBiliStream
    MessageSegment(type='text', data={'text':textMsg}),
  File "/usr/lib/python3.8/typing.py", line 873, in __new__
    obj = super().__new__(cls)
TypeError: Can't instantiate abstract class MessageSegment with abstract methods __str__, get_message_class, is_text

不知道应该怎么办，来论坛请教一下QAQ

**Comment by yanyongyu** [comment, marked as answer]
请从你使用的adapter import `Message`哦，比如

```python
from nonebot.adapters.onebot.v11 import Message, MessageSegment
```

**Reply by TDK1969** [reply]
已解决，非常感谢！
同时还想请教一下，能否从本地发送图片？因为通过url发送图片尺寸过大，可能需要在本地压缩一下~
![image](https://user-images.githubusercontent.com/54183084/155330761-8b27e68a-15b7-4f84-a1e7-c5d0655820ac.png)

**Reply by yanyongyu** [reply]
参考onebot协议格式发送本地文件 https://github.com/botuniverse/onebot-11/blob/master/message/segment.md#%E5%9B%BE%E7%89%87

或者直接使用pathlib传文件

```python
form pathlib import Path
MessageSegment.image(Path("/path/to/image"))
```
